File: pset1/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int min_num_coins(int cents);

int main(void)
{
    float change_owed;
    do
    {
        //Ask how much change is owed.
        change_owed = get_float("Change owed: ");
    }
    while (change_owed < 0);
    //Convert to cents
    int cents = round(change_owed * 100);

    //Compute min of coins possible
    int res = min_num_coins(cents);

    //Print min of coins possible with\n
    printf("%i\n", res);
}

int min_num_coins(int cents)
//Compute min of coins possible on owned cents
{
    int cnt = 0;
    while (cents > 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
        }
        else if (cents >= 10)
        {
            cents -= 10;
        }
        else if (cents >= 5)
        {
            cents -= 5;
        }
        else
        {
            cents--;
        }
        cnt++;
    }
    return cnt;
}

File: pset6/cash/cash.py
"""cash.py
Calculates the minimum number of coins required to give a user change."""


def main():
    while True:
        try:
            # Prompt user
            owed = float(input("Change owed: "))
            # Check non-negative
            if owed >= 0:
                break
        except ValueError:
            pass
    
    # Convert to cents
    cents = round(owed * 100)

    # Compute min of coins required
    res = min_coins(cents)

    # Print result
    print(res)


def min_coins(n: float) -> int:
    """ Compute min of coins possible on owned cents"""
    cnt = 0
    while (n > 0):
        if n >= 25:
            n -= 25
        elif n >= 10:
            n -= 10
        elif n >= 5:
            n -= 5
        else:
            n -= 1
        cnt += 1
    return cnt


if __name__ == "__main__":
    main()
